gravitee-am-test/specs/gateway/mfa.jest.spec.ts

```diff
@@ -0,0 +1,689 @@
+/*
+ * Copyright (C) 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {afterAll, beforeAll, expect, jest} from "@jest/globals";
+import {createDomain, deleteDomain, startDomain} from "@management-commands/domain-management-commands";
+import {requestAdminAccessToken} from "@management-commands/token-management-commands";
+import {createApplication, updateApplication} from "@management-commands/application-management-commands";
+
+import fetch from "cross-fetch";
+import {buildCreateAndTestUser, deleteUser} from "@management-commands/user-management-commands";
+import {createFactor} from "@management-commands/factor-management-commands";
+import {createResource} from "@management-commands/resource-management-commands";
+import {extractXsrfTokenAndActionResponse, getWellKnownOpenIdConfiguration, logoutUser, performFormPost, performGet, performPost} from "@gateway-commands/oauth-oidc-commands";
+import {clearEmails, getLastEmail} from "@utils-commands/email-commands";
+
+const cheerio = require('cheerio');
+
+global.fetch = fetch;
+
+let domain;
+let accessToken;
+let smsFactor;
+let smsResource;
+let smtpResource;
+let emailFactor;
+let openIdConfiguration;
+let smsTestApp;
+let recoveryCodeFactor;
+let recoveryCodeTestApp;
+let bruteForceTestApp;
+let emailTestApp;
+
+const mfaChallengeAttemptsResetTime = 1
+const validMFACode = '333333'
+const sharedSecret = "K546JFR2PK5CGQLLUTFG4W46IKDFWWUE";
+//this verificationCode is generated with the sharedSecret above. Do not change the value unless required
+const emailVerificationCode = "346147";
+
+jest.setTimeout(200000)
+
+beforeAll(async () => {
+    const adminTokenResponse = await requestAdminAccessToken();
+    accessToken = adminTokenResponse.body.access_token;
+    expect(accessToken).toBeDefined();
+    domain = await createDomain(accessToken, "mfa-test-domain", "test mfa");
+    expect(domain).toBeDefined();
```

Comment:
You should really be making assertions in set up code, furthermore I don't think it adds anything to the test to assert whether the domain has been created, whether it has an ID etc as the test would fail were that not true.

Reply:
yep, removed the unnecessary assertions